tests/src/PatternAssertion/CarouselPatternAssert.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_bootstrap_theme\PatternAssertion;
+
+use Symfony\Component\DomCrawler\Crawler;
+
+/**
+ * Assertions for the carousel pattern.
+ */
+class CarouselPatternAssert extends BasePatternAssert {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getAssertions(string $variant): array {
+    return [
+      'items' => [
+        [$this, 'assertItems'],
+      ],
+      'settings' => [
+        [$this, 'assertSettings'],
+      ],
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function assertBaseElements(string $html, string $variant): void {
+    $crawler = new Crawler($html);
+
+    $this->assertElementExists('.carousel .carousel-inner', $crawler);
+  }
+
+  /**
+   * Asserts the carousel pattern settings.
+   *
+   * @param array[] $expected
+   *   The expected settings.
+   * @param \Symfony\Component\DomCrawler\Crawler $crawler
+   *   The crawler.
+   */
+  protected function assertSettings(array $expected, Crawler $crawler): void {
+    if (!empty($expected['fade'])) {
+      $this->assertElementExists('.carousel-fade', $crawler);
+    }
+    else {
+      $this->assertElementNotExists('.carousel-fade', $crawler);
+    }
+
+    if (!empty($expected['show_controls'])) {
+      $this->assertElementText('Previous', '.carousel-control-prev', $crawler);
```

Comment:
Most of these methods can receive NULL and they will check for absence of element/attribute.
We could rewrite (some of) them like this:
```suggestion
      $this->assertElementText($expected['show_controls'] ?? NULL, '.carousel-control-prev', $crawler);
```